WordPress/Classes/Services/NotificationSyncService.swift

```diff
@@ -22,12 +22,20 @@ class NotificationSyncService
     ///
     private let maximumNotes = 100
 
-    /// Returns the main CoredAta Context
+    /// Main CoreData Context
     ///
     private var mainContext: NSManagedObjectContext {
         return contextManager.mainContext
     }
 
+    /// Thread Safety Helper!
+    ///
+    private static let lock = NSLock()
+
+    /// Shared PrivateContext among all of the Sync Service Instances
+    ///
+    private static var privateContext: NSManagedObjectContext!
```

Comment:
Ughhhh this feels so nasty. One of the founding principles of the service architecture in the app is no state is retained between method calls.